src/arch/riscv/kernel/vspace.c

```diff
@@ -37,6 +39,13 @@
 #include <plat/machine/hardware.h>
 #include <kernel/stack.h>
 
+
+typedef enum PAGETABLE_LEVEL_E {
+        PT_LEVEL_1 = 1,
+        PT_LEVEL_2 = 2,
+        PT_LEVEL_3 = 3,
+} ptlevel_t;
+
```

Comment:
Can I ask what the intended use is for this enum? The PT design for RISC-V is generic in the number of levels, i.e. the only level one *should* need a symbolic name for is the top level, the rest should be computed. This is different to how it works on all other platforms in seL4, but since RISC-V is very regular, we're trying to take advantage of that in the (ongoing) verification. If at all possible, I'd very much like to remove the temptation for people to talk about specific levels elsewhere in the code.

I can see that this might be useful in platform-specific init code, but since PT_LEVEL_2 really is just 2, I think it'd be justified to not use an enum for it.

Reply:
The purpose of the enum was to avoid the use of "magic numbers". However, we definitely aren't as familiar with how the RISC-V code is structured. We can definitely remove the use of this enum in the kernel code, since the code is already structured to be "SV Independent". 

I can especially see when compiling for 32-bit mode how `PT_LEVEL_1` would be confusing.

Reply:
Since you don't want levels discussed in the code, I also renamed the top level array to `kernel_toplevel_pageTable`

Reply:
nitpick: ```kernel_root_pageTable``` would also follow both riscv and seL4 conventions. ```level1``` is also fine as it exists for all Sv modes

Reply:
I updated to `kernel_toplevel_pageTable`

Reply:
sorry @heshamelmatary, I didn't see that you had commented on this change. I'll update to `root_pageTable` as that makes the most sense.